### PR TITLE
Initial troubleshooting guide

### PR DESCRIPTION
**Pull request by lulf**
### Type of change

- Documentation

### Description

Initial guide with troubleshooting steps.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/modules/proc-troubleshooting.adoc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get addressspace _myspace_ -o wide
+----
+
+. Check address space controller logs for errors:
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} logs dc/address-space-controller
```
Comment (robshelly) [inline documentation/modules/proc-troubleshooting.adoc]: Should this be `{cmdcli} logs deployment/address-space-controller`?

File: documentation/modules/proc-troubleshooting.adoc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get address _myspace.myaddress_ -o wide
+----
+
+. If using `brokered` address space, check `agent` pod logs for errors:
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} logs dc/agent._uuid_
```
Comment (robshelly) [inline documentation/modules/proc-troubleshooting.adoc]: Same as above, should this be `deployment` instead deployment config?

File: documentation/modules/proc-troubleshooting.adoc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+----
+
+. If using `standard` address space, check `admin` pod logs for errors:
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} logs dc/admin._uuid_ -c standard-controller
+{cmdcli} logs dc/admin._uuid_ -c agent
```
Comment (robshelly) [inline documentation/modules/proc-troubleshooting.adoc]: Same as above, should these be `deployment` instead deployment config?

**Comment by lulf** [comment]
@robshelly Indeed! Updated...

File: documentation/modules/proc-troubleshooting.adoc [hunk 4]
```diff
@@ -0,0 +1,105 @@
+[id='troubleshooting-{context}']
+
+= Find which pods belong to an address space
+
+.Procedure
+
+. Retrieve the uuid for the address space
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Retrieve the uuid for the address space:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get addressspace _myspace_ -o jsonpath='{.metadata.annotations.enmasse\.io/infra-uuid}'
+----
+
+. Find all pods with matching `infraUuid` label:
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Find all pods with a matching `infraUuid` label:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get pods -l infraUuid=_uuid_
+----
+
+= Find currently running version of {ProductName}
+
+.Procedure
+
+. Find enmasse-operator pod name:
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Determine the `enmasse-operator` pod name:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get pods | grep enmasse-operator
+----
+
+. Print the VERSION environment of the enmasse-operator pod:
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Print the `VERSION` environment of the `enmasse-operator` pod:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} rsh _enmasse-operator-...._ env | grep VERSION
+----
+
+The VERSION should correspond to the Major.Minor version of the product, whereas MAVEN_VERSION corresponds to the build number.
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
+
The `VERSION` corresponds to the major.minor version of {ProductName}, whereas `MAVEN_VERSION` corresponds to the build number.
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+----
+
+. Find all pods with matching `infraUuid` label:
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get pods -l infraUuid=_uuid_
+----
+
+= Find currently running version of {ProductName}
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
= Determine the currently running version of {ProductName}
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+. Print the VERSION environment of the enmasse-operator pod:
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} rsh _enmasse-operator-...._ env | grep VERSION
+----
+
+The VERSION should correspond to the Major.Minor version of the product, whereas MAVEN_VERSION corresponds to the build number.
+
+= Unable to find custom address space plan
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
= Unable to find a custom address space plan
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+.Procedure 
+
+. Check that the address space plan is created in the namespace where {ProductName} is installed.
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Ensure that the address space plan was created in the namespace where {ProductName} is installed. 
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+----
+
+The VERSION should correspond to the Major.Minor version of the product, whereas MAVEN_VERSION corresponds to the build number.
+
+= Unable to find custom address space plan
+
+.Procedure 
+
+. Check that the address space plan is created in the namespace where {ProductName} is installed. 
+. Check that the address space plan has passed validation and is in the `Active` phase:
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Ensure that the address space plan has passed validation and is in the `Active` phase:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get addressplans -o wide -n {ProductNamespace}
+----
+
+= Address space is not ready and clients are unable to connect to messaging endpoints
+
+.Procedure
+. Check if address space status message for errors:
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Check the address space status message for errors:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get addressspaceplans -o wide -n {ProductNamespace}
+----
+
+= Unable to find custom address plan
+
+.Procedure 
+
+. Check that the address plan is created in the namespace where {ProductName} is installed.
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Ensure that the address plan was created in the namespace where {ProductName} is installed. 
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+
+. Check that the address space plan is created in the namespace where {ProductName} is installed. 
+. Check that the address space plan has passed validation and is in the `Active` phase:
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get addressspaceplans -o wide -n {ProductNamespace}
+----
+
+= Unable to find custom address plan
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
= Unable to find a custom address plan
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+
+.Procedure 
+
+. Check that the address plan is created in the namespace where {ProductName} is installed. 
+. Check that the address plan has passed validation and is in the `Active` phase:
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Ensure that the address plan has passed validation and is in the `Active` phase:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+
+.Procedure
+. Check if address status message for errors:
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} get address _myspace.myaddress_ -o wide
+----
+
+. If using `brokered` address space, check `agent` pod logs for errors:
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. If using the `brokered` address space, check the `agent` pod logs for errors:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} logs deployment/agent._uuid_
+----
+
+. If using `standard` address space, check `admin` pod logs for errors:
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
+
If using the `standard` address space, check the `admin` pod logs for errors:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
++
+[options="nowrap",subs="+quotes,attributes"]
+----
+{cmdcli} logs deployment/address-space-controller
+----
+
+= Addresses are not getting ready
+
+.Procedure
+. Check if address status message for errors:
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
. Check the address status message for errors:
```

File: documentation/modules/proc-troubleshooting.adoc [hunk 20]
```diff
@@ -0,0 +1,105 @@
+[id='troubleshooting-{context}']
+
+= Find which pods belong to an address space
```
Comment (jenmalloy) [inline documentation/modules/proc-troubleshooting.adoc]:
```suggestion
= Determine which pods belong to an address space
```

**Pull request merged**